src/Extensions/Compatability/Validation/Code.cs

```diff
@@ -0,0 +1,515 @@
+		[Obsolete(ObsoleteMessage, IsObsoleteError)]
+		public static bool Validate(bool state, string errorMessage, EventId? tagId)
+		{
+			if (!state)
+			{
+				LogError(tagId, errorMessage);
+			}
+
+			return state;
+		}
+
+
+		/// <summary>
+		/// Validate object's state
+		/// </summary>
+		/// <param name="state">Object's state</param>
+		/// <param name="errorMessage">Error message</param>
+		/// <param name="tagId">Tag Id to log, leave null if no logging is needed</param>
+		/// <param name="parameters">logging parameters</param>
+		/// <returns>True if state evaluates to true; false otherwise</returns>
+		[Obsolete(ObsoleteMessage, IsObsoleteError)]
+		public static bool Validate(bool state, EventId? tagId, string errorMessage, params object[] parameters)
+		{
+			if (!state)
+			{
+				LogError(tagId, errorMessage, parameters);
+			}
+
+			return state;
+		}
+
+
+		/// <summary>
+		/// Checks that the collection is not null, is not empty, and does not contain nulls
+		/// </summary>
+		/// <typeparam name="T">The type of the collection</typeparam>
+		/// <param name="argumentValue">The argument value.</param>
+		/// <param name="argumentName">Name of the argument.</param>
+		/// <param name="tagId">Tag Id to log, leave null if no logging is needed</param>
+		/// <returns>True if the argument <paramref name="argumentValue"/> is not null, is not empty, and does not contain nulls; false otherwise.</returns>
+		[Obsolete(ObsoleteMessage, IsObsoleteError)]
+		public static bool ValidateNotEmptyAndAllNotNull<T>([NotNullWhen(true)]ICollection<T>? argumentValue, string argumentName, EventId? tagId = null)
+			where T : class
+		{
+			if (!ValidateArgument(argumentValue, argumentName, tagId))
+			{
+				return false;
+			}
+
+			if (!argumentValue.Any() || argumentValue.Any(x => x == null))
+			{
+				LogError(tagId, ValidationFailed, AllErrorMessage, argumentName);
+				return false;
+			}
+
+			return true;
+		}
+
+
+		/// <summary>
+		/// Checks that the enumerable argument is not null and doesn't contain any nulls
+		/// </summary>
+		/// <remarks>Be careful to not pass enumerables that can be enumerated only once</remarks>
+		/// <typeparam name="TEnumerable">Type of the enumerable</typeparam>
+		/// <param name="argumentValue">The argument value.</param>
+		/// <param name="argumentName">Name of the argument.</param>
+		/// <param name="tagId">Tag Id to log, leave null if no logging is needed</param>
+		/// <returns>True if the argument <paramref name="argumentValue"/> is not null and contains only non-null elements; false otherwise.</returns>
+		[Obsolete(ObsoleteMessage, IsObsoleteError)]
+		public static bool ValidateAllNotNull<TEnumerable>([NotNullWhen(true)]IEnumerable<TEnumerable>? argumentValue, string argumentName, EventId? tagId = null)
+			where TEnumerable : class
+		{
+			if (!ValidateArgument(argumentValue, argumentName, tagId))
+			{
+				return false;
+			}
+
+			if (argumentValue.Any(x => x == null))
```

Comment:
Why is the logic here slightly different than in the method above?

Reply:
explained here https://github.com/microsoft/Omex/pull/153#discussion_r390509059